test/Shared/UserManagerTestBase.cs

```diff
@@ -592,8 +591,8 @@ protected TRole CreateRole(string namePrefix = "")
             {
                 IdentityResultAssert.IsSuccess(await mgr.CreateAsync(u));
             }
-            var usersQ = mgr.Users.Where(u => u.UserName.StartsWith("CanFindUsersViaUserQuerable"));
-            Assert.Null(mgr.Users.FirstOrDefault(u => u.UserName == "bogus"));
+            var usersQ = mgr.Users.Where(u => mgr.GetUserNameAsync(u).Result.StartsWith("CanFindUsersViaUserQuerable"));
+            Assert.Null(mgr.Users.FirstOrDefault(u => mgr.GetUserNameAsync(u).Result == "bogus"));
```

Comment:
sure. The `.Result` was something i din't like too but ran out of ideas to make it better since the variable was inside the lambda. We can chat about it.